starky/src/proof.rs

```diff
@@ -55,6 +57,40 @@ pub struct StarkProofTarget<const D: usize> {
 }
 
 impl<const D: usize> StarkProofTarget<D> {
+    /// Serializes a STARK proof.
+    pub fn to_buffer(&self, buffer: &mut Vec<u8>) -> IoResult<()> {
```

Comment:
`buffer` could be slice I think

Reply:
Actually it can't, the `Write` trait for serialization is only implemented over `Vec<u8>`

Reply:
Ugh. That's a bit surprising and annoying. Oh well, not this PR for sure... :)